src/components/ProfileTree/ProfileTree.js

```diff
@@ -83,33 +95,26 @@ export const ProfileTree = memo(({
     setInitialData(newData);
   };
 
-  const unlink = row => {
-    const index = data.findIndex(item => item.id === row.id);
-    const newIdx = findRelIndex(relationsToAdd, row);
+  const unlink = (parentData, setParentData, line, masterId, masterType, detailType, reactTo, localDataKey) => {
+    const index = parentData.findIndex(item => item.id === line.id);
+    const newIdx = findRelIndex(relationsToAdd, line);
 
     if (newIdx < 0) {
-      const relsToDel = [...relationsToDelete, ...composeRelations([row])];
+      const newRels = composeRelations([line], masterId, masterType, detailType, reactTo);
+      const relsToDel = [...relationsToDelete, ...newRels];
 
       onUnlink(relsToDel);
     }
 
-    data.splice(index, 1);
-    setData(data);
-    sessionStorage.setItem(dataKey, JSON.stringify(data));
-  };
-
-  const remove = row => {
-    const index = data.findIndex(item => item.id === row.id);
-    const newIdx = findRelIndex(relationsToAdd, row);
+    const newData = [...parentData];
 
-    if (newIdx < 0) {
-      const relsToDel = [...relationsToDelete, ...composeRelations([row])];
-
-      onUnlink(relsToDel);
-    }
+    newData.splice(index, 1);
+    setParentData(newData);
+    sessionStorage.setItem(localDataKey, JSON.stringify(newData));
+  };
 
-    data.splice(index, 1);
-    setData(data);
+  const remove = (parentData, setParentData, line, masterId, masterType, detailType, reactTo, localDataKey) => {
```

Comment:
Do we really need this functionality? As far as I can see in the story description, there is only `unlink` action.

Reply:
Yes as far as this functionality has been discussed and will be needed in the future.